root/event/edit_form.tt

```diff
@@ -58,5 +60,25 @@
 <script type="text/javascript">
   (function () {
     MB.Control.initialize_guess_case("event", "id-edit-event");
+
+    function blockTooShortBeginYear() {
+      var beginYear = $('#id-edit-event\\\.period\\\.begin_date\\\.year').val();
+      var allowed = (beginYear === null || beginYear === '' || beginYear.length === 4);
+      $('.submit').prop('disabled', !allowed);
+      $('#too_short_begin_year').toggle(!allowed);
+    }
+
+    function blockTooShortEndYear() {
+      var endYear = $('#id-edit-event\\\.period\\\.end_date\\\.year').val();
+      var allowed = (endYear === null || endYear === '' || endYear.length === 4);
+      $('.submit').prop('disabled', !allowed);
+      $('#too_short_end_year').toggle(!allowed);
+    }
+
+    $('#id-edit-event\\\.period\\\.begin_date\\\.year').keyup(blockTooShortBeginYear).change(blockTooShortBeginYear);
+    blockTooShortBeginYear();
+
+    $('#id-edit-event\\\.period\\\.end_date\\\.year').keyup(blockTooShortEndYear).change(blockTooShortEndYear);
+    blockTooShortEndYear();
```

Comment:
A macro which takes the entity `type` as an argument and outputs this script would be nicer than duplicating it 3x (here, root/artist/edit_form.tt, and root/components/forms.tt).